v2_task.go

```diff
@@ -22,6 +24,56 @@ type TaskDateTime struct {
 	time.Time
 }
 
+// Duration extends the standard time.Duration type to provide specialized
+// unmarshaling from JSON. Unlike time.Duration, which typically requires
+// duration values to be specified in string format (like "300ms", "1h30m"),
+// Duration is designed to unmarshal from two distinct JSON value types:
+//
+//  1. Integer values representing the duration in seconds.
+//  2. A specific string literal "< 1", which we interpret as half a second.
+//
+// Examples:
+//   - `{"duration": 60}` unmarshals to 60 seconds,
+//   - `{"duration": "< 1"}` unmarshals to 500 milliseconds.
+//
+// Python function: https://github.com/artefactual/archivematica/blob/2c03a79e710d655c3740ddbc4bc62eebd4e4b4fe/src/dashboard/src/components/helpers.py#L163-L170.
+type Duration time.Duration
+
+type Task struct {
+	ID          string       `json:"uuid"`
+	ExitCode    uint8        `json:"exit_code"`
+	FileID      string       `json:"file_uuid"`
+	Filename    string       `json:"file_name"`
+	CreatedAt   TaskDateTime `json:"time_created"`
+	StartedAt   TaskDateTime `json:"time_started"`
+	CompletedAt TaskDateTime `json:"time_ended"`
+	Duration    Duration     `json:"duration"`
+}
+
+func (d *Duration) UnmarshalJSON(data []byte) error {
+	// Check for the special "< 1" case.
+	if bytes.ContainsRune(data, '<') {
```

Comment:
I suggested `bytes.Equal(data, []byte("\"< 1\""))` because we're looking for that specific string. In this version you've written, `BROKEN<STRING` would not raise an error, my version however returned:

    strconv.ParseInt: parsing "\"<fwef 1\"": invalid syntax

We should restore this condition:

```suggestion
	if bytes.Equal(data, []byte("\"< 1\"")) {
```

Reply:
But it is not possible for the API to return this response.